rust/cloud-storage/macro_db_client/src/user/get_user_name.rs

```diff
@@ -47,3 +47,243 @@ pub async fn get_user_names(
 
     Ok(user_names)
 }
+
+/// gets macro user names for passed user profile ids, falling back to the user's email contacts if they have any
+#[tracing::instrument(skip(db))]
+pub async fn get_user_names_with_email(
+    db: &sqlx::PgPool,
+    macro_user_id: &str,
+    user_profile_ids: &Vec<String>,
```

Comment:
This should not be a string and instead MacroUserId<Lowercase<'_>>

You should either use a `NonEmpty<Vec<MacroUserId<Lowercase<'_>>>` here to ensure that there are values to be checked or if you want to go the old-school approach add in a check at the start of this method that checks if user_profile_ids is empty